### PR TITLE
Features/guyber/add signals results to emulator

### PR DESCRIPTION
**Pull request by bermanguy**
Start implementation of results page.

### CODE REVIEW

File: src/SDK/SmartSignalsRuntimeAppEmulator/Models/SmartSignalRunner.cs [hunk 1]
```diff
@@ -102,16 +121,38 @@ public bool IsSignalRunning
         public async Task RunAsync(List<ResourceIdentifier> resources, TimeSpan analysisCadence)
         {
             this.cancellationTokenSource = new CancellationTokenSource();
-            this.Result = null;
+            this.Results.Clear();
             var analysisRequest = new AnalysisRequest(resources, null, analysisCadence, this.analysisServicesFactory);
             try
             {
+                // Run Signl
```
Comment (liadtal) [inline src/SDK/SmartSignalsRuntimeAppEmulator/Models/SmartSignalRunner.cs]: >Signl [](start = 23, length = 5)

Signal

**Comment by EranGon** [comment]
>                 .RegisterInstance(signalRunner);

why don't you use RegisterType instead? this way you don't have to create all of these objects here...

---
Refers to: src/SDK/SmartSignalsRuntimeAppEmulator/App.xaml.cs:80 in 1f19eaa. [](commit_id = 1f19eaa75d53fe1aa0771339d94f13c41efdc193, deletion_comment = False)

**Comment by EranGon** [comment]
>                 .RegisterInstance(signalRunner);

(you'll still need to use RegisterInstance for the tracer, but that's it)

---
In reply to: [364730571](https://github.com/liadtal/Smart-Alerts/pull/72#issuecomment-364730571) [](ancestors = 364730571)

---
Refers to: src/SDK/SmartSignalsRuntimeAppEmulator/App.xaml.cs:80 in 1f19eaa. [](commit_id = 1f19eaa75d53fe1aa0771339d94f13c41efdc193, deletion_comment = False)

File: src/SDK/SmartSignalsRuntimeAppEmulator/Models/AuthenticationServices.cs [hunk 2]
```diff
@@ -66,9 +66,9 @@ public void AuthenticateUser()
                 ResourceId,
                 ClientId,
                 RedirectUri,
-                PromptBehavior.Auto,
-                UserIdentifier.AnyUser, 
-                "prompt=consent");
+                PromptBehavior.Auto);
+            ////UserIdentifier.AnyUser, 
+            ////"prompt=consent");
```
Comment (EranGon) [inline src/SDK/SmartSignalsRuntimeAppEmulator/Models/AuthenticationServices.cs]: ??

File: src/SDK/SmartSignalsRuntimeAppEmulator/Controls/SignalResultsControl.xaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            <Setter Property="Margin" Value="5,0,0,0"/>
+            <Setter Property="Background" Value="White"/>
+        </Style>
+        <Style TargetType="{x:Type ListView}">
+            <Style.Resources>
+                <Style TargetType="{x:Type GridViewColumnHeader}" BasedOn="{StaticResource ResultsColumnHeaderStyle}" />
+            </Style.Resources>
+        </Style>
+    </UserControl.Resources>
+    <Grid Background="#FFECF0F1">
```
[the remark this answers is not in the document]
Comment (bermanguy) [reply src/SDK/SmartSignalsRuntimeAppEmulator/Controls/SignalResultsControl.xaml]: it is the background we always use...

---
In reply to: [167424059](https://github.com/liadtal/Smart-Alerts/pull/72#discussion_r167424059) [](ancestors = 167424059)

File: src/SDK/SmartSignalsRuntimeAppEmulator/Models/AuthenticationServices.cs (hunk 2 above)
Comment (bermanguy) [reply src/SDK/SmartSignalsRuntimeAppEmulator/Models/AuthenticationServices.cs]: :O

---
In reply to: [167424108](https://github.com/liadtal/Smart-Alerts/pull/72#discussion_r167424108) [](ancestors = 167424108)

**Pull request merged**